app/src/main/java/com/cronus/zdone/api/TasksRepositoryImpl.kt

```diff
@@ -1,78 +1,104 @@
 package com.cronus.zdone.api
 
 import com.cronus.zdone.AppExecutors
-import com.cronus.zdone.WorkTimeManager
 import com.cronus.zdone.api.model.*
 import com.cronus.zdone.home.HomeScreen
 import io.reactivex.Observable
+import io.reactivex.ObservableEmitter
+import java.util.*
 import javax.inject.Inject
 import javax.inject.Singleton
+import kotlin.collections.HashSet
 
 @Singleton
 class TasksRepositoryImpl @Inject constructor(
-        var appExecutors: AppExecutors,
-        var zdoneService: ZdoneService,
-        var workTimeManager: WorkTimeManager
+    var appExecutors: AppExecutors,
+    var zdoneService: ZdoneService
 ) : TasksRepository {
 
-    private val CACHE_REFRESH_TIME = 2 * 60 * 1_000_000_000L
+    private val CACHE_REFRESH_TIME = 2 * 60 * 1_000_000_000L // 2 mins in nanoseconds
```

Comment:
Do we need this refresh time? Can we not just depend on cache being invalidated by any edit? I guess we still have to implement the server-push stuff for mobile -- where you edit something on desktop and would like to have it reflect on mobile. Do you have some docs on where I should be pushing the updates? Perhaps using https://firebase.google.com/docs/cloud-messaging/?

Reply:
yeah we will want to keep this around until we get push notifications working. I think firebase cloud messenger (link you posted) is the way to go. Should be the simplest/most reliable solution as it is Google-approved. Hopefully it doesn't go the way of Google Reader tho...